collector/filesystem_common.go

```diff
@@ -0,0 +1,113 @@
+// +build !nofilesystem
```

Comment:
I believe we need to add another build tag:

`// +build linux freebsd`

Otherwise, `filesystem_common.go` would be attempted to compile but it would fail on anything but linux or freebsd. (`*_linux.go` and `*_freebsd.go` are implicitly only compiled on the respective OSs.)